### ISSUE TITLE
Alignment doesn't work in Jupyter due to joblib

### ISSUE THREAD

**Issue by sdwfrost** [veg/BioExt#27]
If I try to align sequences in a Jupyter notebook, I get an error, due to the dependency on joblib. Could you implement a serial version of `_align_par`?

**Comment by stevenweaver** [comment]
Yes, I think this should be straightforward to do by specifying a different backend for joblib.Parallel. Would it be possible to provide the offending workbook?

**Comment by stevenweaver** [comment]
Dear Simon, 

I've removed the local, modified version of joblib in BioExt and modified our code to use the packaged distribution of joblib. This appears to have resolved the issue. 

Please see the following branch
https://github.com/veg/BioExt/tree/gh-27

I've confirmed that the following code does run in a jupyter notebook.


```
import os
import nose
import functools
import os
from Bio import SeqIO

from BioExt.io import BamIO

from BioExt.args import (
    add_alphabet,
    add_reference,
    add_scorematrix
    )

from BioExt.uds import _align_par
from BioExt.misc import compute_cigar, gapless

from BioExt.scorematrices import (
    DNAScoreMatrix,
    FrequenciesError,
    ProteinScoreMatrix,
    BLOSUM62
    )

## Load reference sequence
seqpath = "./rsrc/SHORT.FASTA"
output_file = "./rsrc/SHORT.FASTA.test.bam"

records = SeqIO.parse(seqpath, 'fasta')

reference = gapless(next(records))

def allseqs(records):
    yield compute_cigar(reference, reference)
    for record in records:
        print(record)
        yield record

def output(records):
    BamIO.write(
        allseqs(records),
        output_file,
        reference
        )   

_align_par(
    reference,
    records,
    BLOSUM62.load(),
    True,
    False,
    None,
    None,
    output,
    False
    )   


# Read output file
BamIO.sort(output_file)
```

I did set the argument for `require='sharedmem'` in joblib.Parallel. This may result in slightly slower performance. 

Best,
Steven

**Comment by stevenweaver** [comment]
Update : I will include gh-27 in the next release after I've completed benchmarking the changes.

**Comment by stevenweaver** [comment]
Fix in bioext>=0.19.3.
